internal/provider/atlas_schema_resource.go

```diff
@@ -299,83 +311,77 @@ func PrintPlanSQL(ctx context.Context, data *AtlasSchemaResourceModel) (diags di
+	defer func() {
+		if err := cleanup(); err != nil {
+			tflog.Debug(ctx, "Failed to remove HCL file", map[string]interface{}{
+				"error": err,
+			})
+		}
+	}()
+	_, err = r.client.SchemaApply(ctx, &atlas.SchemaApplyParams{
+		DevURL:  data.DevURL.Value,
+		Exclude: exclude,
+		To:      to,
+		URL:     data.URL.Value,
+	})
+	if err != nil {
+		diags.AddError("Apply Error",
+			fmt.Sprintf("Unable to apply changes, got error: %s", err),
+		)
 		return
 	}
 	return diags
 }
 
-func firstRunCheck(ctx context.Context, data *AtlasSchemaResourceModel) (diags diag.Diagnostics) {
-	createDesired := func(ctx context.Context, cli *sqlclient.Client) (desired *schema.Realm, err error) {
-		desired = &schema.Realm{}
-		p := hclparse.NewParser()
-		if _, err := p.ParseHCL([]byte(data.HCL.Value), ""); err != nil {
-			return nil, err
-		}
-		if err = cli.Evaluator.Eval(p, desired, nil); err != nil {
-			return
-		}
+func (r *AtlasSchemaResource) firstRunCheck(ctx context.Context, data *AtlasSchemaResourceModel) (diags diag.Diagnostics) {
+	to, cleanup, err := data.handleHCL()
+	if err != nil {
+		diags.AddError("HCL Error",
+			fmt.Sprintf("Unable to create temporary file for HCL, got error: %s", err),
+		)
 		return
 	}
-	changes, cli, diags := atlasChanges(ctx, data, createDesired)
+	defer func() {
+		if err := cleanup(); err != nil {
+			tflog.Debug(ctx, "Failed to remove HCL file", map[string]interface{}{
+				"error": err,
+			})
+		}
+	}()
+	var exclude []string
+	diags.Append(data.GetExclude(ctx, &exclude)...)
 	if diags.HasError() {
 		return
 	}
-	defer cli.Close()
-
+	result, err := r.client.SchemaApply(ctx, &atlas.SchemaApplyParams{
+		DevURL:  data.DevURL.Value,
+		DryRun:  true,
+		Exclude: exclude,
+		To:      to,
+		URL:     data.URL.Value,
+	})
+	if err != nil {
+		diags.AddError("Atlas Plan Error",
+			fmt.Sprintf("Unable to generate migration plan, got error: %s", err),
+		)
+		return
+	}
 	var causes []string
-	for _, c := range changes {
-		switch c := c.(type) {
-		case *schema.DropSchema:
-			causes = append(causes, fmt.Sprintf("DROP SCHEMA %q", c.S.Name))
-		case *schema.DropTable:
-			causes = append(causes, fmt.Sprintf("DROP TABLE %q", c.T.Name))
-		case *schema.ModifyTable:
-			for _, c1 := range c.Changes {
-				switch t := c1.(type) {
-				case *schema.DropColumn:
-					causes = append(causes, fmt.Sprintf("DROP COLUMN %q.%q", c.T.Name, t.C.Name))
-				case *schema.DropIndex:
-					causes = append(causes, fmt.Sprintf("DROP INDEX %q.%q", c.T.Name, t.I.Name))
-				case *schema.DropForeignKey:
-					causes = append(causes, fmt.Sprintf("DROP FOREIGN KEY %q.%q", c.T.Name, t.F.Symbol))
-				case *schema.DropAttr:
-					causes = append(causes, fmt.Sprintf("DROP ATTRIBUTE %q.%T", c.T.Name, t.A))
-				case *schema.DropCheck:
-					causes = append(causes, fmt.Sprintf("DROP CHECK CONSTRAINT %q.%q", c.T.Name, t.C.Name))
-				}
-			}
+	for _, c := range result.Changes.Pending {
+		if strings.HasPrefix(c, "DROP ") {
```

Comment:
Column/indexes/etc drops are part of ALTER TABLE in some databases. Why not leave it as it was before? You don't have access this info?